lb4 datasource --config results in an error - "function () { [native code] }" is generated

## Steps to reproduce

create a new data source with --config
`lb4 datasource -c '{"name": "dbname","connector": "postgresql", "url": "", "host": "psql-svc", "port": "5432", "user": "psql", "password": "pw", "database": "dbname"}' -y` 
then start the lb4 app:
`npm start`

creates errors like:
```src/datasources/test.datasource.ts:37:36 - error TS1005: ',' expected.

37         log: function () { [native code] },
```

When I use the exact same values through using the prompt, there are no issues when starting the lb4 app.


## Current Behavior

creates errors like:
```src/datasources/test.datasource.ts:37:36 - error TS1005: ',' expected.

37         log: function () { [native code] },
                                      ~~~~

src/datasources/test.datasource.ts:38:37 - error TS1005: ',' expected.

38         warn: function () { [native code] },
                                       ~~~~

src/datasources/test.datasource.ts:39:36 - error TS1005: ',' expected.

39         dir: function () { [native code] },
                                      ~~~~

src/datasources/test.datasource.ts:40:37 - error TS1005: ',' expected.

40         time: function () { [native code] },
                                       ~~~~
```
etc...


## Expected Behavior

The lb4 application should start with no errors - as when using the prompt process.
Interestingly, creating the datasource works when using 


## Additional information

`$ node -e 'console.log(process.platform, process.arch, process.versions.node)'
linux x64 16.13.0
`

`$ npm ls --prod --depth 0 | grep loopback
+-- @loopback/boot@4.0.0
+-- @loopback/cli@3.0.0
+-- @loopback/core@3.0.0
+-- @loopback/repository@4.0.0
+-- @loopback/rest-explorer@4.0.0
+-- @loopback/rest@11.0.0
+-- @loopback/service-proxy@4.0.0`

## Related Issues

Many issues reported that a prompt appears - but it doesn't when using the -y flag.
But haven't found anyone reporting the issue that the data source doesn't work when created using --config.

--- a/packages/cli/generators/controller/index.js
+++ b/packages/cli/generators/controller/index.js
@@ -86,6 +86,13 @@ module.exports = class ControllerGenerator extends ArtifactGenerator {
       utils.toClassName(this.artifactInfo.name),
     );
 
+    if (this.options.controllerType) {
+      Object.assign(this.artifactInfo, {
+        controllerType: this.options.controllerType,
+      });
+      return;
+    }
+
     return this.prompt([
       {
         type: 'list',

--- a/packages/cli/generators/datasource/index.js
+++ b/packages/cli/generators/datasource/index.js
@@ -80,6 +80,11 @@ module.exports = class DataSourceGenerator extends ArtifactGenerator {
   promptArtifactName() {
     debug('Prompting for artifact name');
     if (this.shouldExit()) return false;
+
+    if (this.artifactInfo.name) {
+      return;
+    }
+
     const prompts = [
       {
         type: 'input',

--- a/packages/cli/generators/discover/index.js
+++ b/packages/cli/generators/discover/index.js
@@ -109,6 +109,20 @@ module.exports = class DiscoveryGenerator extends ArtifactGenerator {
         path.resolve(dsDir, `${utils.toFileName(s)}.datasource.js`),
       ),
     );
+
+    if (this.options.dataSource) {
+      if (
+        this.dataSourceChoices
+          .map(d => d.name)
+          .includes(this.options.dataSource)
+      ) {
+        Object.assign(this.artifactInfo, {
+          dataSource: this.dataSourceChoices.find(
+            d => d.name === this.options.dataSource,
+          ),
+        });
+      }
+    }
     debug(`Done importing datasources`);
   }
 

--- a/packages/cli/generators/example/index.js
+++ b/packages/cli/generators/example/index.js
@@ -84,6 +84,10 @@ module.exports = class extends BaseGenerator {
     return super._setupGenerator();
   }
 
+  setOptions() {
+    return super.setOptions();
+  }
+
   help() {
     const examplesHelp = Object.keys(EXAMPLES)
       .map(name => `  ${name}: ${EXAMPLES[name]}`)

--- a/packages/cli/generators/extension/index.js
+++ b/packages/cli/generators/extension/index.js
@@ -43,6 +43,13 @@ module.exports = class ExtensionGenerator extends ProjectGenerator {
 
   promptComponent() {
     if (this.shouldExit()) return;
+
+    if (this.options.componentName) {
+      Object.assign(this.projectInfo, {
+        componentName: this.options.componentName,
+      });
+    }
+
     const prompts = [
       {
         type: 'input',

--- a/packages/cli/generators/openapi/index.js
+++ b/packages/cli/generators/openapi/index.js
@@ -90,6 +90,10 @@ module.exports = class OpenApiGenerator extends BaseGenerator {
     return super._setupGenerator();
   }
 
+  setOptions() {
+    return super.setOptions();
+  }
+
   checkLoopBackProject() {
     return super.checkLoopBackProject();
   }

--- a/packages/cli/generators/repository/index.js
+++ b/packages/cli/generators/repository/index.js
@@ -426,6 +426,7 @@ module.exports = class RepositoryGenerator extends ArtifactGenerator {
         name: 'repositoryBaseClass',
         message: PROMPT_BASE_REPOSITORY_CLASS,
         choices: availableRepositoryList,
+        when: this.artifactInfo.repositoryBaseClass === undefined,
         default:
           this.artifactInfo.repositoryBaseClass === undefined
             ? availableRepositoryList[0]

--- a/packages/cli/lib/artifact-generator.js
+++ b/packages/cli/lib/artifact-generator.js
@@ -46,6 +46,12 @@ module.exports = class ArtifactGenerator extends BaseGenerator {
   promptArtifactName() {
     debug('Prompting for artifact name');
     if (this.shouldExit()) return false;
+
+    if (this.options.name) {
+      Object.assign(this.artifactInfo, {name: this.options.name});
+      return;
+    }
+
     const prompts = [
       {
         type: 'input',

